### ISSUE TITLE
🐛 git's --color-moved with DOS line endings (\r\n) triggers a panic

### ISSUE THREAD

**Issue by th1000s** [dandavison/delta#795]
Using delta 92d6591e and git version 2.30.2. To reproduce:

    $ echo -e "move_this_long_function()\\nfoo()\\nbar()" | sed 's,$,\r,' > a.py
    $ { tail -n 2 a.py; head -n 1 a.py; } > b.py 
    $ git diff --color-moved --no-index --color=always a.py b.py | target/release/delta --no-gitconfig 
    
    thread 'main' panicked at 'String mismatch encountered while superimposing style sections: '
    '', src/paint.rs:921:17
    
    [..]
    2: delta::paint::superimpose_style_sections::superimpose_style_sections
    3: delta::paint::Painter::paint_line
    4: delta::paint::Painter::paint_buffered_minus_and_plus_lines
    5: delta::delta::delta
    6: delta::main
    $ cat -A a.py
    move_this_long_function()^M$
    foo()^M$
    bar()^M$

**Comment by dandavison** [comment]
Thanks @th1000s. This is happening because git (for some reason -- does anyone know why? Is it intentional?) inserts ANSI escape sequences between the \r and \n (in this case a "reset" command). This behavior of git was noted in https://github.com/dandavison/delta/pull/664.


```
$ git diff --color-moved --no-index --color=always a.py b.py | bat -pA
␛[1mdiff·--git·a/a.py·b/b.py␛[m␊
␛[1mindex·cb0f8bb..3dc21ec·100644␛[m␊
␛[1m---·a/a.py␛[m␊
␛[1m+++·b/b.py␛[m␊
␛[36m@@·-1,3·+1,3·@@␛[m␊
␛[1;35m-move_this_long_function()␛[m␍␊
·foo()␛[m␍␊
·bar()␛[m␍␊
␛[1;36m+␛[m␛[1;36mmove_this_long_function()␛[m␛[41m␍␛[m␊
```

Fixed in #800 by doing the \r-stripping on the raw line, accounting for the possibility of trailing ANSI escape sequences.

**Comment by dandavison** [comment]
My guess is this is a bug in git -- failing to account for the possibility of \r\n endings when inserting a reset ANSI code, but terminal emulators tolerate it.
